build/xtask/src/dist.rs

```diff
@@ -1290,7 +1317,7 @@ fn check_task_priorities(toml: &Config) -> Result<()> {
 
 fn generate_task_linker_script(
     name: &str,
-    map: &BTreeMap<String, Range<u32>>,
+    map: &BTreeMap<String, Vec<Range<u32>>>,
```

Comment:
Does the code below assume this vec is sorted? I notice it accesses first and last.

Reply:
Same as below, fixed with `ContiguousRanges` (3d128cbee0fd75fab17799a754281d836d4a7321)